src/System.Text.Primitives/System/Text/Parsing/InvariantUtf16/InvariantUtf16_bool.cs

```diff
@@ -12,13 +12,50 @@ public static partial class InvariantUtf16
         {
             public unsafe static bool TryParseBoolean(char* text, int length, out bool value)
             {
-                ReadOnlySpan<char> span = new ReadOnlySpan<char>(text, length);
-                return TryParseBoolean(span, out value);
+                value = default(bool);
+                if (length < 1)
```

Comment:
Would it be better to remove this check? It's not even accurate anymore, i.e. should be < 4. But then IsTrue does the same check again.